Hyperbolic_surface_triangulation_2/include/CGAL/Hyperbolic_surface_triangulation_2.h

```diff
@@ -0,0 +1,721 @@
+
+  Hyperbolic_surface_triangulation_2() {};
+  Hyperbolic_surface_triangulation_2(const Hyperbolic_fundamental_domain_2<Traits>& domain);
+  Hyperbolic_surface_triangulation_2(const Combinatorial_map_with_cross_ratios& cmap);
+  Hyperbolic_surface_triangulation_2(const Combinatorial_map_with_cross_ratios& cmap, const Anchor& anchor);
+
+  //Hyperbolic_surface_triangulation_2& operator=(Hyperbolic_surface_triangulation_2&& other);
+  Hyperbolic_surface_triangulation_2& operator=(Hyperbolic_surface_triangulation_2 other);
+
+  Combinatorial_map_with_cross_ratios& get_combinatorial_map_ref();
+  bool has_anchor() const;
+  Anchor& get_anchor_ref();
+
+  void to_stream(std::ostream& s) const;
+  void from_stream(std::istream& s);
+
+  bool is_delaunay_flippable(Dart_handle dart) const;
+  void flip(Dart_handle dart);
+  bool is_delaunay() const;
+  int make_delaunay();
+  std::vector<std::tuple<Dart_const_handle,Point,Point,Point>> lift(bool center=true) const;
+
+  bool is_valid() const;
+
+private:
+  Combinatorial_map_with_cross_ratios _combinatorial_map;
+  bool _has_anchor = false;
+  Anchor _anchor;
+
+  Dart_handle ccw(Dart_handle dart);
+  Dart_handle cw(Dart_handle dart);
+  Dart_handle opposite(Dart_handle dart);
+  Dart_const_handle const_ccw(Dart_const_handle dart) const;
+  Dart_const_handle const_cw(Dart_const_handle dart) const;
+  Dart_const_handle const_opposite(Dart_const_handle dart) const;
+
+  ComplexNumber get_cross_ratio(Dart_const_handle dart) const;
+
+  Dart_handle pick_edge_to_flip();
+
+  void copy_from(const Combinatorial_map_with_cross_ratios& cmap);
+  void copy_from(const Combinatorial_map_with_cross_ratios& cmap, const Anchor& anchor);
+
+  // Returns the cross ratio of the points a,b,c,d
+  ComplexNumber cross_ratio(const Point& a, const Point& b, const Point& c, const Point& d) const;
+  // Returns the point d such that the cross ratio of a,b,c,d is cratio
+  Point fourth_point_from_cross_ratio(const Point& a, const Point& b, const Point& c, const ComplexNumber& cratio) const;
+};
+
+template<class Traits> std::ostream& operator<<(std::ostream& s, Hyperbolic_surface_triangulation_2<Traits>& Hyperbolic_surface_triangulation_2);
+template<class Traits> void operator>>(std::istream& s, Hyperbolic_surface_triangulation_2<Traits>& triangulation);
+
+////////////////////////////////////////////////////////////////////////////////
+////////////////////////////////////////////////////////////////////////////////
+////////////////////////////////////////////////////////////////////////////////
+
+template<class Traits>
+Hyperbolic_surface_triangulation_2<Traits>::Hyperbolic_surface_triangulation_2(const Domain& domain){
+  // (Triangulates by adding an internal edge between domain.vertex(size-1) and the other vertices)
+  _combinatorial_map.clear();
+  int size = domain.size();
+
+  // Make the triangles
+  Dart_handle dart_of_triangle[size-2];
```

Comment:
You must use a `std::vector`.   A built-in array VC++ does not allow a size known at runtime. I just fixed that.